### PR TITLE
FIX: optimizing feature creation from text

### PR DESCRIPTION
**Pull request by razmikmelikbekyan**
*Issue #, if available:*

*Description of changes:*
Basi PEP8 corrections, 
Correction of Type checking, 
Vectorization of feature creations from text,
Removing duplicate len calculation from strings.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: autogluon/utils/tabular/features/abstract_feature_generator.py [hunk 1]
```diff
@@ -350,18 +359,16 @@ def check_if_nlp_feature(self, X: Series):
 
     def generate_text_features(self, X: Series, feature: str) -> DataFrame:
         X: DataFrame = X.to_frame(name=feature)
-        X[feature + '.char_count'] = [self.char_count(value) for value in X[feature].values]
-        X[feature + '.word_count'] = [self.word_count(value) for value in X[feature].values]
-        X[feature + '.capital_ratio'] = [self.capital_ratio(value) for value in X[feature].values]
-        X[feature + '.lower_ratio'] = [self.lower_ratio(value) for value in X[feature].values]
-        X[feature + '.digit_ratio'] = [self.digit_ratio(value) for value in X[feature].values]
-        X[feature + '.special_ratio'] = [self.special_ratio(value) for value in X[feature].values]
+        X[feature + '.char_count'] = X[feature].apply(self.char_count)
```
Comment (Innixma) [inline autogluon/utils/tabular/features/abstract_feature_generator.py]: How does this affect performance? I remember .apply() having bad performance in pandas which is why I generally avoided it for most things (like using zip list comprehensions as a replacement)

**Comment by szha** [comment]
Job PR-300-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-300/1/index.html

File: autogluon/utils/tabular/features/abstract_feature_generator.py (hunk 1 above)
Comment (razmikmelikbekyan) [reply autogluon/utils/tabular/features/abstract_feature_generator.py]: Hey Nick, 
You are right, pandas.apply performs slightly worse than list comprehension. Check the notebook: https://github.com/razmikmelikbekyan/ML-playground/blob/master/miscellaneous/Performance%20Checks.ipynb
So I will revert them back to list comprehensions.

**Comment by szha** [comment]
Job PR-300-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-300/2/index.html

File: autogluon/utils/tabular/features/abstract_feature_generator.py (hunk 1 above)
Comment (Innixma) [reply autogluon/utils/tabular/features/abstract_feature_generator.py]: Thanks for checking!

**Pull request merged**